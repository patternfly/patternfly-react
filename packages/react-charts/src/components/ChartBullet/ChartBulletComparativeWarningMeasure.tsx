import * as React from 'react';
import hoistNonReactStatics from 'hoist-non-react-statics';
import { DataGetterPropType, DomainPropType, NumberOrCallback, PaddingProps } from 'victory-core';
import { VictoryBar } from 'victory-bar';
import { ChartContainer } from '../ChartContainer';
import { ChartThemeDefinition } from '../ChartTheme';
import { getBulletComparativeWarningMeasureTheme } from '../ChartUtils';
import { ChartBulletComparativeMeasure } from './ChartBulletComparativeMeasure';

/**
 * See https://github.com/FormidableLabs/victory/blob/master/packages/victory-core/src/index.d.ts
 * and https://github.com/FormidableLabs/victory/blob/master/packages/victory-bar/src/index.d.ts
 */
export interface ChartBulletComparativeWarningMeasureProps {
  /**
   * Specifies the tooltip capability of the container component. A value of true allows the chart to add a
   * ChartTooltip component to the labelComponent property. This is a shortcut to display tooltips when the labels
   * property is also provided.
   */
  allowTooltip?: boolean;
  /**
   * The ariaDesc prop specifies the description of the chart/SVG to assist with
   * accessibility for screen readers.
   *
   * Note: Overridden by the desc prop of containerComponent
   */
  ariaDesc?: string;
  /**
   * The ariaTitle prop specifies the title to be applied to the SVG to assist
   * accessibility for screen readers.
   *
   * Note: Overridden by the title prop of containerComponent
   */
  ariaTitle?: string;
  /**
   * The barWidth prop is used to specify the width of each bar. This prop may be given as
   * a number of pixels or as a function that returns a number. When this prop is given as
   * a function, it will be evaluated with the arguments datum, and active. When this value
   * is not given, a default value will be calculated based on the overall dimensions of
   * the chart, and the number of bars.
   */
  barWidth?: NumberOrCallback;
  /**
   * The constrainToVisibleArea prop determines whether to coerce tooltips so that they fit within the visible area of
   * the chart. When this prop is set to true, tooltip pointers will still point to the correct data point, but the
   * center of the tooltip will be shifted to fit within the overall width and height of the svg Victory renders.
   */
  constrainToVisibleArea?: boolean;
  /**
   * The data prop specifies the data to be plotted. Data should be in the form of an array
   * of data points, or an array of arrays of data points for multiple datasets.
   * Each data point may be any format you wish (depending on the `comparativeErrorMeasureDataY` accessor prop),
   * but by default, an object with y properties is expected.
   *
   * @example data={[{ y: 50 }]}
   */
  data?: any[];
  /**
   * The domain prop describes the range of values your chart will cover. This prop can be
   * given as a array of the minimum and maximum expected values for your bar chart,
   * or as an object that specifies separate arrays for x and y.
   * If this prop is not provided, a domain will be calculated from data, or other
   * available information.
   *
   * @example {x: [0, 2], y: [0, 100]}
   *
   * Note: The x domain is expected to be `x: [0, 2]` in order to position all measures properly
   */
  domain?: DomainPropType;
  /**
   * The height props specifies the height the svg viewBox of the chart container.
   * This value should be given as a number of pixels
   */
  height?: number;
  /**
   * The horizontal prop determines whether data will be plotted horizontally.
   * When this prop is set to true, the independent variable will be plotted on the y axis
   * and the dependent variable will be plotted on the x axis.
   */
  horizontal?: boolean;
  /**
   * The labelComponent prop takes in an entire label component which will be used
   * to create a label for the bar. The new element created from the passed labelComponent
   * will be supplied with the following properties: x, y, index, data, verticalAnchor,
   * textAnchor, angle, style, text, and events. any of these props may be overridden
   * by passing in props to the supplied component, or modified or ignored within
   * the custom component itself. If labelComponent is omitted, a new ChartLabel
   * will be created with props described above. This labelComponent prop should be used to
   * provide a series label for ChartBar. If individual labels are required for each
   * data point, they should be created by composing ChartBar with VictoryScatter
   */
  labelComponent?: React.ReactElement<any>;
  /**
   * The labels prop defines labels that will appear above each bar in your chart.
   * This prop should be given as an array of values or as a function of data.
   * If given as an array, the number of elements in the array should be equal to
   * the length of the data array. Labels may also be added directly to the data object
   * like data={[{y: 1, label: "first"}]}.
   *
   * @example ["spring", "summer", "fall", "winter"], (datum) => datum.title
   */
  labels?: string[] | number[] | ((data: any) => string | number | null);
  /**
   * The measureComponent prop takes an entire component which will be used to create the chart
   */
  measureComponent?: React.ReactElement<any>;
  /**
   * The padding props specifies the amount of padding in number of pixels between
   * the edge of the chart and any rendered child components. This prop can be given
   * as a number or as an object with padding specified for top, bottom, left
   * and right.
   */
  padding?: PaddingProps;
  /**
   * The standalone prop determines whether the component will render a standalone svg
   * or a <g> tag that will be included in an external svg. Set standalone to false to
   * compose ChartLine with other components within an enclosing <svg> tag.
   */
  standalone?: boolean;
  /**
   * The theme prop takes a style object with nested data, labels, and parent objects.
   * You can create this object yourself, or you can use a theme provided by
   * When using ChartLine as a solo component, implement the theme directly on
   * ChartLine. If you are wrapping ChartLine in ChartChart or ChartGroup,
   * please call the theme on the outermost wrapper component instead.
   */
  theme?: ChartThemeDefinition;
  /**
   * Specifies the theme color. Valid values are 'blue', 'green', 'multi', etc.
   *
   * Note: Not compatible with theme prop
   *
   * @example themeColor={ChartThemeColor.blue}
   */
  themeColor?: string;
  /**
   * Specifies the theme variant. Valid values are 'dark' or 'light'
   *
   * Note: Not compatible with theme prop
   *
   * @example themeVariant={ChartThemeVariant.light}
   */
  themeVariant?: string;
  /**
   * The width props specifies the width of the svg viewBox of the chart container
   * This value should be given as a number of pixels
   */
  width?: number;
  /**
   * The y prop specifies how to access the Y value of each data point.
   * If given as a function, it will be run on each data point, and returned value will be used.
   * If given as an integer, it will be used as an array index for array-type data points.
   * If given as a string, it will be used as a property key for object-type data points.
   * If given as an array of strings, or a string containing dots or brackets,
   * it will be used as a nested object property path (for details see Lodash docs for _.get).
   * If `null` or `undefined`, the data value will be used as is (identity function/pass-through).
   *
   * @example 0, 'y', 'y.value.nested.1.thing', 'y[2].also.nested', null, d => Math.sin(d)
   */
  y?: DataGetterPropType;
}

export const ChartBulletComparativeWarningMeasure: React.FunctionComponent<
  ChartBulletComparativeWarningMeasureProps
> = ({
  allowTooltip = true,
  ariaDesc,
  ariaTitle,
  barWidth,
  constrainToVisibleArea = false,
  data,
  domain,
  horizontal = true,
  labelComponent,
  labels,
  measureComponent = <ChartBulletComparativeMeasure />,
  padding,
  standalone = true,
  themeColor,
  themeVariant,
  y,

  // destructure last
  theme = getBulletComparativeWarningMeasureTheme(themeColor, themeVariant),
  height = theme.bar.height,
  width = theme.bar.width
}: ChartBulletComparativeWarningMeasureProps) => {
  // Comparative measure component
  const measure = React.cloneElement(measureComponent, {
    allowTooltip,
    ariaDesc,
    ariaTitle,
    barWidth,
    constrainToVisibleArea,
    data,
    domain,
    height,
    horizontal,
    labelComponent,
    labels,
    padding,
    standalone: false,
    theme,
    width,
    y,
    ...measureComponent.props
  });

  return standalone ? (
    <ChartContainer desc={ariaDesc} height={height} title={ariaTitle} width={width}>
      {measure}
    </ChartContainer>
  ) : (
    <React.Fragment>{measure}</React.Fragment>
  );
};

// Note: VictoryBar.role must be hoisted
hoistNonReactStatics(ChartBulletComparativeWarningMeasure, VictoryBar);
