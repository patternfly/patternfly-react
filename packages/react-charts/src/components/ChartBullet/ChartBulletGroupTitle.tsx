import * as React from 'react';
import { PaddingProps, Line, StringOrNumberOrCallback } from 'victory-core';
import { ChartContainer } from '../ChartContainer';
import { ChartLabel } from '../ChartLabel';
import { ChartBulletStyles, ChartThemeDefinition } from '../ChartTheme';
import {
  getBulletGroupTitleTheme,
  getLabelTextSize,
  getBulletLabelX,
  getBulletLabelY,
  getPaddingForSide
} from '../ChartUtils';

/**
 * See https://github.com/FormidableLabs/victory/blob/master/packages/victory-core/src/index.d.ts
 */
export interface ChartBulletGroupTitleProps {
  /**
   * The ariaDesc prop specifies the description of the chart/SVG to assist with
   * accessibility for screen readers.
   *
   * Note: Overridden by the desc prop of containerComponent
   */
  ariaDesc?: string;
  /**
   * The ariaTitle prop specifies the title to be applied to the SVG to assist
   * accessibility for screen readers.
   *
   * Note: Overridden by the title prop of containerComponent
   */
  ariaTitle?: string;
  /**
   * The capHeight prop defines a text metric for the font being used: the expected height of capital letters.
   * This is necessary because of SVG, which (a) positions the *bottom* of the text at `y`, and (b) has no notion of
   * line height. The value should ideally use the same units as `lineHeight` and `dy`, preferably ems. If given a
   * unitless number, it is assumed to be ems.
   */
  capHeight?: StringOrNumberOrCallback;
  /**
   * The divider component to render with the chart
   */
  dividerComponent?: React.ReactElement<any>;
  /**
   * The height props specifies the height the svg viewBox of the chart container.
   * This value should be given as a number of pixels
   */
  height?: number;
  /**
   * The padding props specifies the amount of padding in number of pixels between
   * the edge of the chart and any rendered child components. This prop can be given
   * as a number or as an object with padding specified for top, bottom, left
   * and right.
   *
   * Note: The bottom padding property is unused
   */
  padding?: PaddingProps;
  /**
   * The standalone prop determines whether the component will render a standalone svg
   * or a <g> tag that will be included in an external svg. Set standalone to false to
   * compose ChartLine with other components within an enclosing <svg> tag.
   */
  standalone?: boolean;
  /**
   * The subtitle to render for grouped bullets
   */
  subTitle?: string;
  /**
   * The label component to render the chart subTitle.
   *
   * Note: Default label properties may be applied
   */
  subTitleComponent?: React.ReactElement<any>;
  /**
   * The theme prop takes a style object with nested data, labels, and parent objects.
   * You can create this object yourself, or you can use a theme provided by
   * When using ChartLine as a solo component, implement the theme directly on
   * ChartLine. If you are wrapping ChartLine in ChartChart or ChartGroup,
   * please call the theme on the outermost wrapper component instead.
   */
  theme?: ChartThemeDefinition;
  /**
   * Specifies the theme color. Valid values are 'blue', 'green', 'multi', etc.
   *
   * Note: Not compatible with theme prop
   *
   * @example themeColor={ChartThemeColor.blue}
   */
  themeColor?: string;
  /**
   * Specifies the theme variant. Valid values are 'dark' or 'light'
   *
   * Note: Not compatible with theme prop
   *
   * @example themeVariant={ChartThemeVariant.light}
   */
  themeVariant?: string;
  /**
   * The title to render for grouped bullets
   */
  title?: string;
  /**
   * The label component to render the chart title.
   *
   * Note: Default label properties may be applied
   */
  titleComponent?: React.ReactElement<any>;
  /**
   * The width props specifies the width of the svg viewBox of the chart container
   * This value should be given as a number of pixels
   */
  width?: number;
}

export const ChartBulletGroupTitle: React.FunctionComponent<ChartBulletGroupTitleProps> = ({
  ariaDesc,
  ariaTitle,
  capHeight = 1.1,
  dividerComponent = <Line />,
  padding,
  standalone = true,
  subTitle,
  themeColor,
  themeVariant,
  title,
  titleComponent = <ChartLabel />,

  // destructure last
  theme = getBulletGroupTitleTheme(themeColor, themeVariant),
  height = theme.chart.height,
  width = theme.chart.width
}: ChartBulletGroupTitleProps) => {
  const defaultPadding = {
    bottom: getPaddingForSide('bottom', padding, theme.chart.padding),
    left: getPaddingForSide('left', padding, theme.chart.padding),
    right: getPaddingForSide('right', padding, theme.chart.padding),
    top: getPaddingForSide('top', padding, theme.chart.padding)
  };

  const labelPadding = {
    bottom: getPaddingForSide('bottom', padding, Number(theme.legend.style.labels.padding)),
    left: getPaddingForSide('left', padding, Number(theme.legend.style.labels.padding)),
    right: getPaddingForSide('right', padding, Number(theme.legend.style.labels.padding)),
    top: getPaddingForSide('top', padding, Number(theme.legend.style.labels.padding))
  };

  // Horizontal divider to render under the group title
  const getDivider = () => {
    const titleSize = getLabelTextSize({ text: title, theme });
    const subTitleSize = getLabelTextSize({ text: subTitle, theme });
    const dy =
      title && subTitle
        ? titleSize.height + subTitleSize.height + labelPadding.top + labelPadding.bottom
        : titleSize.height + labelPadding.top + labelPadding.bottom;

    return React.cloneElement(dividerComponent, {
      x1: defaultPadding.left,
      x2: width - defaultPadding.right,
      y1: defaultPadding.top + dy,
      y2: defaultPadding.top + dy,
      style: theme.line.style.data,
      ...dividerComponent.props
    });
  };

  // Returns title
  const getTitle = () => {
    const titleProps = titleComponent ? titleComponent.props : {};
    const showBoth = title && subTitle;
    return React.cloneElement(titleComponent, {
      ...(showBoth && { capHeight }),
      style: [ChartBulletStyles.label.groupTitle, ChartBulletStyles.label.subTitle],
      text: showBoth ? [title, subTitle] : title,
      textAnchor: 'middle',
      verticalAnchor: 'middle',
      x: getBulletLabelX({
        chartWidth: width,
        labelPosition: 'top'
      }),
      y: getBulletLabelY({
        chartHeight: height,
        dy: defaultPadding.top,
        labelPosition: 'top'
      }),
      ...titleProps
    });
  };

  const groupTitle = Boolean(title) && (
    <React.Fragment>
      {getTitle()}
      {getDivider()}
    </React.Fragment>
  );

  return standalone ? (
    <ChartContainer desc={ariaDesc} height={height} title={ariaTitle} width={width}>
      {groupTitle}
    </ChartContainer>
  ) : (
    <React.Fragment>{groupTitle}</React.Fragment>
  );
};
