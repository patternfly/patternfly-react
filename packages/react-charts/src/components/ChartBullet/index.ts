export * from './ChartBullet';
export * from './ChartBulletComparativeErrorMeasure';
export * from './ChartBulletComparativeMeasure';
export * from './ChartBulletComparativeWarningMeasure';
export * from './ChartBulletPrimaryDotMeasure';
export * from './ChartBulletPrimarySegmentedMeasure';
export * from './ChartBulletQualitativeRange';
