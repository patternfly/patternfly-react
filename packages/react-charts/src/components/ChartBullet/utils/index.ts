export * from './chart-bullet-data';
export * from './chart-bullet-domain';
export * from './chart-bullet-size';
export * from './chart-bullet-theme';
