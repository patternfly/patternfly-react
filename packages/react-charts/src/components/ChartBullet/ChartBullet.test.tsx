import * as React from 'react';
import { render } from '@testing-library/react';
import { ChartBullet } from './ChartBullet';

Object.values([true, false]).forEach(() => {
  test('ChartBulletQualitativeRange', () => {
    const view = render(<ChartBullet />);
    expect(view.container).toMatchSnapshot();
  });
});

test('renders component data', () => {
  const view = render(
    <ChartBullet
      ariaDesc="Storage capacity"
      ariaTitle="Bullet chart example"
      comparativeWarningMeasureData={[{ name: 'Warning', y: 88 }]}
      labels={({ datum }) => `${datum.name}: ${datum.y}`}
      maxDomain={{ y: 100 }}
      primarySegmentedMeasureData={[{ name: 'Measure', y: 50 }]}
      qualitativeRangeData={[
        { name: 'Range', y: 50 },
        { name: 'Range', y: 75 }
      ]}
      width={450}
    />
  );
  expect(view.container).toMatchSnapshot();
});
