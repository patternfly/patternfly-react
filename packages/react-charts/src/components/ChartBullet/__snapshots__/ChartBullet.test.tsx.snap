// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartBulletQualitativeRange 1`] = `
<DocumentFragment>
  <div
    class="pf-c-chart"
    style="pointer-events: none; position: relative; width: 100%; height: 100%;"
  >
    <svg
      height="140"
      role="img"
      style="pointer-events: all; width: 100%; height: 100%;"
      viewBox="0 0 450 140"
      width="450"
    >
      <g
        role="presentation"
      >
        <line
          role="presentation"
          shape-rendering="auto"
          style="stroke: #d2d2d2; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
          vector-effect="non-scaling-stroke"
          x1="50"
          x2="400"
          y1="85"
          y2="85"
        />
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
            vector-effect="non-scaling-stroke"
            x1="50"
            x2="50"
            y1="85"
            y2="90"
          />
          <text
            direction="inherit"
            dx="0"
            id="axis-tickLabels-0"
            x="50"
            y="111.97"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
              text-anchor="middle"
              x="50"
            >
              0
            </tspan>
          </text>
        </g>
      </g>
      <g>
        <rect
          height="0"
          role="presentation"
          shape-rendering="auto"
          style="fill: none;"
          vector-effect="non-scaling-stroke"
          width="0"
          x="225"
          y="131"
        />
      </g>
    </svg>
    <div
      style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
    >
      <svg
        height="140"
        style="overflow: visible; width: 100%; height: 100%;"
        viewBox="0 0 450 140"
        width="450"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ChartBulletQualitativeRange 2`] = `
<DocumentFragment>
  <div
    class="pf-c-chart"
    style="pointer-events: none; position: relative; width: 100%; height: 100%;"
  >
    <svg
      height="140"
      role="img"
      style="pointer-events: all; width: 100%; height: 100%;"
      viewBox="0 0 450 140"
      width="450"
    >
      <g
        role="presentation"
      >
        <line
          role="presentation"
          shape-rendering="auto"
          style="stroke: #d2d2d2; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
          vector-effect="non-scaling-stroke"
          x1="50"
          x2="400"
          y1="85"
          y2="85"
        />
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
            vector-effect="non-scaling-stroke"
            x1="50"
            x2="50"
            y1="85"
            y2="90"
          />
          <text
            direction="inherit"
            dx="0"
            id="axis-tickLabels-0"
            x="50"
            y="111.97"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
              text-anchor="middle"
              x="50"
            >
              0
            </tspan>
          </text>
        </g>
      </g>
      <g>
        <rect
          height="0"
          role="presentation"
          shape-rendering="auto"
          style="fill: none;"
          vector-effect="non-scaling-stroke"
          width="0"
          x="225"
          y="131"
        />
      </g>
    </svg>
    <div
      style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
    >
      <svg
        height="140"
        style="overflow: visible; width: 100%; height: 100%;"
        viewBox="0 0 450 140"
        width="450"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders component data 1`] = `
<DocumentFragment>
  <div
    class="pf-c-chart"
    style="pointer-events: none; position: relative; width: 100%; height: 100%;"
  >
    <svg
      aria-describedby="victory-container-3-desc"
      aria-labelledby="victory-container-3-title"
      height="140"
      role="img"
      style="pointer-events: all; width: 100%; height: 100%;"
      viewBox="0 0 450 140"
      width="450"
    >
      <title
        id="victory-container-3-title"
      >
        Bullet chart example
      </title>
      <desc
        id="victory-container-3-desc"
      >
        Storage capacity
      </desc>
      <g
        role="presentation"
      >
        <line
          role="presentation"
          shape-rendering="auto"
          style="stroke: #d2d2d2; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
          vector-effect="non-scaling-stroke"
          x1="50"
          x2="400"
          y1="85"
          y2="85"
        />
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
            vector-effect="non-scaling-stroke"
            x1="50"
            x2="50"
            y1="85"
            y2="90"
          />
          <text
            direction="inherit"
            dx="0"
            id="axis-tickLabels-0"
            x="50"
            y="111.97"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
              text-anchor="middle"
              x="50"
            >
              0
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
            vector-effect="non-scaling-stroke"
            x1="400"
            x2="400"
            y1="85"
            y2="90"
          />
          <text
            direction="inherit"
            dx="0"
            id="axis-tickLabels-1"
            x="400"
            y="111.97"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
              text-anchor="middle"
              x="400"
            >
              100
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
            vector-effect="non-scaling-stroke"
            x1="137.5"
            x2="137.5"
            y1="85"
            y2="90"
          />
          <text
            direction="inherit"
            dx="0"
            id="axis-tickLabels-2"
            x="137.5"
            y="111.97"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
              text-anchor="middle"
              x="137.5"
            >
              25
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
            vector-effect="non-scaling-stroke"
            x1="225"
            x2="225"
            y1="85"
            y2="90"
          />
          <text
            direction="inherit"
            dx="0"
            id="axis-tickLabels-3"
            x="225"
            y="111.97"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
              text-anchor="middle"
              x="225"
            >
              50
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
            vector-effect="non-scaling-stroke"
            x1="312.5"
            x2="312.5"
            y1="85"
            y2="90"
          />
          <text
            direction="inherit"
            dx="0"
            id="axis-tickLabels-4"
            x="312.5"
            y="111.97"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
              text-anchor="middle"
              x="312.5"
            >
              75
            </tspan>
          </text>
        </g>
      </g>
      <g
        role="presentation"
      >
        <path
          d="M 50, 85 
A 0 0 0 0 1, 50, 85 
L 50, 55 
A 0 0 0 0 1, 50, 55 
L 312.5, 55 
A 0 0 0 0 1, 312.5, 55 
L 312.5, 85 
A 0 0 0 0 1, 312.5, 85 
 z"
          index="0"
          role="presentation"
          shape-rendering="auto"
          style="fill: #d2d2d2; stroke: none; padding: 8px; stroke-width: 0;"
        />
      </g>
      <g
        role="presentation"
      >
        <path
          d="M 50, 85 
A 0 0 0 0 1, 50, 85 
L 50, 55 
A 0 0 0 0 1, 50, 55 
L 225, 55 
A 0 0 0 0 1, 225, 55 
L 225, 85 
A 0 0 0 0 1, 225, 85 
 z"
          index="0"
          role="presentation"
          shape-rendering="auto"
          style="fill: #f0f0f0; stroke: none; padding: 8px; stroke-width: 0;"
        />
      </g>
      <g
        role="presentation"
      >
        <path
          d="M 50, 74.5 
A 0 0 0 0 1, 50, 74.5 
L 50, 65.5 
A 0 0 0 0 1, 50, 65.5 
L 225, 65.5 
A 0 0 0 0 1, 225, 65.5 
L 225, 74.5 
A 0 0 0 0 1, 225, 74.5 
 z"
          index="0"
          role="presentation"
          shape-rendering="auto"
          style="fill: #06c; stroke: none; padding: 8px; stroke-width: 0;"
        />
      </g>
      <g
        role="presentation"
      >
        <path
          d="M 358, 85 
A 0 0 0 0 1, 358, 85 
L 358, 55 
A 0 0 0 0 1, 358, 55 
L 358, 55 
A 0 0 0 0 1, 358, 55 
L 358, 85 
A 0 0 0 0 1, 358, 85 
 z"
          index="0"
          role="presentation"
          shape-rendering="auto"
          style="fill: #ec7a08; stroke: #ec7a08; padding: 8px; stroke-width: 2;"
        />
      </g>
      <g>
        <rect
          height="0"
          role="presentation"
          shape-rendering="auto"
          style="fill: none;"
          vector-effect="non-scaling-stroke"
          width="0"
          x="225"
          y="131"
        />
      </g>
    </svg>
    <div
      style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
    >
      <svg
        height="140"
        style="overflow: visible; width: 100%; height: 100%;"
        viewBox="0 0 450 140"
        width="450"
      />
    </div>
  </div>
</DocumentFragment>
`;
